layers/+window-management/exwm/funcs.el

```diff
@@ -118,7 +118,7 @@ The started process is also added to `exwm-autostart-process-list'."
            for autostart-dir = (concat dir "/autostart")
            for file in (when (file-exists-p autostart-dir)
                          (directory-files autostart-dir t ".+\\.desktop$"))
-           (collect (xdg-desktop-read-file file))))
```

Comment:
Change the function body to
```elisp
(cl-loop with autostart-dirs    =  nil
         with autostart-entries =  nil
         for  xdg-dir           in (cons (xdg-config-home) (xdg-config-dirs))
         for  autostart-dir     =  (expand-file-name (concat xdg-dir "/autostart"))
         when (and (file-exists-p autostart-dir)
                   (not (member autostart-dir autostart-dirs)))
           collect autostart-dir into autostart-dirs
           and nconc (mapcar #'xdg-desktop-read-file
                             (directory-files autostart-dir t ".+\\.desktop$")) into autostart-entries
         finally do (print autostart-entries))
```

Reply:
A few changes from the original one:
- replaced `file-exists-p` with `file-directory-p`, which return t only when it's an existing directory
- duplicated autostart directories are filtered out